File: solutions/cpp/perfect-numbers/1/perfect_numbers.cpp
#include "perfect_numbers.h"
#include <stdexcept>

namespace perfect_numbers {

aliquot_type classify(int num){
    int sum = 0;
    if (num<=0) 
        throw std::domain_error("N must be >0.");
    for(int k=1; k<num; k++){
        if (num % k == 0)
            sum += k;}
    if(sum == num)
        return perfect;
    else if (sum > num)
        return abundant;
    else // (sum < num)
        return deficient;
        
}



}  // namespace perfect_numbers

File: solutions/fortran/perfect-numbers/1/perfect_numbers.f90

module perfect_numbers
  implicit none

contains

  character(len=9) function classify(num)
    integer, intent(in) :: num
    integer :: sum, m
    if (num>0) then
      sum = 0
      do m = 1, (num-1)
        if (mod(num, m)==0) then
          sum = sum + m
        end if
      end do
      if (sum == num) then
          classify = "perfect"
      else if (sum > num) then
          classify = "abundant"
      else
          classify = "deficient"
      end if
    else
      classify = "ERROR"
    end if
  end function

end module
